community/modules/scheduler/slinky/main.tf

```diff
@@ -0,0 +1,160 @@
+# Copyright 2025 "Google LLC"
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+locals {
+  cluster_id_parts = split("/", var.cluster_id)
+  cluster_name     = local.cluster_id_parts[5]
+  cluster_location = local.cluster_id_parts[3]
+  project_id       = var.project_id != null ? var.project_id : local.cluster_id_parts[1]
+
+  # Define affinity settings when node pools are specified
+  node_affinity = var.node_pool_names != null
```

Comment:
Instead of introducing `node_affinity` you could simply leverage `node_pool_affinity` to write your logic. `node_affinity` doesn't seem to be required in this file.

One option is:
```
node_pool_affinity = var.node_pool_names != null ? <non null condition as it is> : null
```

You could replace the null with {} as you have done in the current logic and just use `node_pool_affinity!={}`